src/PKSim.Core/Services/PKAnalysesTask.cs

```diff
@@ -878,8 +878,18 @@ public IReadOnlyList<PopulationPKAnalysis> AggregatePKAnalysis(Simulation simula
       {
          var pkParametersList = pkParameters.ToList();
          var matrix = new FloatMatrix();
-         var names = pkParametersList.Select(x => x.Name).Distinct().ToList();
-         pkParametersList.Each(pkParameter => matrix.AddValuesAndSort(pkParameter.ValuesAsArray));
+         var names = new List<string>();
+
+         pkParametersList.Each(pkParameter =>
+         {
+            if (!pkParameter.ValuesAsArray.All(x => x.IsValid())) 
```

Comment:
feels like a tell don't ask, or at least a method
pkParameter.where(hasAtLeastOnVavldValue).Each(

Reply:
Hang on, what I put here is that all the values have to be valid. Do you think this should be changed? In the example there aren't any with 'some' values and 'some' NaN's. Only all-or-nothing.

should I change it to be `if (!pkParameter.ValuesAsArray.Any(x => x.IsValid())) `

Reply:
Oh no. I read it wrong.
I meant just filter and iterate

Reply:
'Any' makes more sense. The method is added.

Reply:
> Oh no. I read it wrong. I meant just filter and iterate

ok